sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -903,6 +910,50 @@ object InferFiltersFromConstraints extends Rule[LogicalPlan]
       Filter(newPredicates.reduce(And), plan)
     }
   }
+
+  /**
```

Comment:
Is this SQL grammar supported by Hive or standard SQL?

Reply:
hive can do this